pycbc/io/hdf.py

```diff
@@ -986,6 +988,91 @@ def to_coinc_xml_object(self, file_name):
+        fap = self.get_coincfile_array('fap')
+        ofd.create_dataset('p_value', data=fap,
+                           dtype=np.float32)
+
+        fap_exc = self.get_coincfile_array('fap_exc')
+        ofd.create_dataset('p_value_exclusive', data=fap_exc,
+                           dtype=np.float32)
+
+        # Coinc fields
+        logging.info("Outputting coinc info")
+        for field in ['stat']:
+            logging.info(field)
+            vals = self.get_coincfile_array(field)
+            ofd.create_dataset(field, data=vals, dtype=np.float32)
+
+        # Bank fields
+        logging.info("Outputting associated bank values")
+        for field in ['mass1','mass2','spin1z','spin2z']:
+            logging.info(field)
+            vals = self.get_bankfile_array(field)
+            ofd.create_dataset(field, data=vals, dtype=np.float32)
+
+        mass1 = self.get_bankfile_array('mass1')
+        mass2 = self.get_bankfile_array('mass2')
+        mchirp, _ = mass1_mass2_to_mchirp_eta(mass1, mass2)
+
+        ofd.create_dataset('chirp_mass', data=mchirp, dtype=np.float32)
+
+        sngl_fields = [f for f in self.sngl_files[self.ifos[0]].columns
+                       if not ('template' in f)]
+        # Single-detector fields
+        logging.info("Outputting single-detector info")
+        for field in sngl_fields:
+            if field in ['gating','search','snr']: continue
+            logging.info(field)
+            vals_valid = self.get_snglfile_array_dict(field)
+            for ifo in self.ifos:
+                vals = vals_valid[ifo][0]
+                valid = vals_valid[ifo][1]
+                vals[np.logical_not(valid)] = -1.
+                ofd.create_dataset(ifo + '_'+field, data=vals,
+                                   dtype=np.float32)
+
+
+        logging.info("Outputting single-detector and network SNRs")
+        snr_vals_valid = self.get_snglfile_array_dict('snr')
+        network_snr_sq = np.zeros_like(snr_vals_valid[self.ifos[0]][0])
+        for ifo in self.ifos:
+            vals = snr_vals_valid[ifo][0]
+            valid = snr_vals_valid[ifo][1]
+            vals[np.logical_not(valid)] = -1.
+            ofd.create_dataset(ifo + '_snr', data=vals,
+                               dtype=np.float32)
+            network_snr_sq[valid] += vals[valid] ** 2.0
+        network_snr = np.sqrt(network_snr_sq)
+        ofd.create_dataset('network_snr', data=network_snr, dtype=np.float32)
+
+        logging.info("Outputting template duration")
+        temp_dur_vals_valid = self.get_snglfile_array_dict('template_duration')
+        temp_dur = np.zeros_like(snr_vals_valid[self.ifos[0]][0])
+        for ifo in self.ifos:
+            vals = temp_dur_vals_valid[ifo][0]
+            valid = temp_dur_vals_valid[ifo][1]
+            temp_dur[valid] = vals[valid]
+        ofd.create_dataset('template_duration', data=temp_dur, dtype=np.float32)
+        logging.info(all(temp_dur > 0))
```

Comment:
This is also a change from the previous PR, as we are getting the template duration. This is a slight faff, as we cant just use one of the ifos as each event wont have triggered in every one, so we have just overwrite this each time